drivers/tee/tee_shm.c

```diff
@@ -48,11 +49,14 @@ static void tee_shm_release(struct tee_shm *shm)
 		poolm->ops->free(poolm, shm);
 	} else if (shm->flags & TEE_SHM_REGISTER) {
 		size_t n;
-		int rc = teedev->desc->ops->shm_unregister(shm->ctx, shm);
+		if (!(shm->flags & TEE_SHM_OCALL)) {
+			rc = teedev->desc->ops->shm_unregister(shm->ctx, shm);
```

Comment:
How about letting the callback `teedev->desc->ops->shm_unregister()` decide for itself what to do if `TEE_SHM_OCALL` is set?

Reply:
That's fair.